### PR TITLE
Add back osx

### PR DESCRIPTION
**Pull request by wolfv**
Migrator has run, but osx is skipped so we don't have a build for macOS ARM

Any reasons for the skipping?

### CODE REVIEW

**Comment by hmaarrfk** [comment]
@wolfv any clue why your cross compilation is failing? x86 host, aarch target.

**Comment by wolfv** [comment]
yeah, in the logs it says:

```
arm64-apple-darwin20.0.0-clang -arch x86_64 -arch x86_64 -arch x86_64 -arch x86_64 -o Darwin19.6.0_arm64-apple-darwin20.0.0-clang_64_OPT.OBJ/utf8.o -c -std=c99 -O2 -fPIC  -fno-common -pipe -DDARWIN -DHAVE_STRERROR -DHAVE_BSD_FLOCK -isysroot /Applications/Xcode_12.4.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.0.sdk -Wall -Wshadow -Qunused-arguments -Wno-parentheses-equality -Wno-array-bounds -Wno-unevaluated-expression -DNSS_NO_GCC48 -DXP_UNIX -UDEBUG -DNDEBUG -DNSS_NO_INIT_SUPPORT -DUSE_UTIL_DIRECTLY -DNO_NSPR_10_SUPPORT -DSSL_DISABLE_DEPRECATED_CIPHER_SUITE_NAMES -I$PREFIX/include/nspr -I$PREFIX/include -I../../../dist/Darwin19.6.0_arm64-apple-darwin20.0.0-clang_64_OPT.OBJ/include -I../../../dist/public/nss -I../../../dist/private/nss  utf8.c
```

where the `-arch x86_64` means it doesn't compile for arm64. I had a build locally on M1 that worked fine... 

Will need to investigate this

**Comment by marcelotrevisani** [comment]
well.. ``osx`` is not my cup of tea, but if you need any help from my side, please let me know :)

**Comment by hmaarrfk** [comment]
It might be guessing the host architecture then. Do you know what was specified for you locally

**Comment by hmaarrfk** [comment]
Can you set ppc64le to program on azure?

**Comment by wolfv** [comment]
yep,  can do. but another fundamental problem is that nss seems to first build a binary (nsinstall) that is later used in the make process to install stuff. However, it doesn't seem to properly work on OS X with cross-compilation enabled... ugh. I can offer to upload a build from a native machine

**Comment by xhochy** [comment]
Seems like you can add the following to the `make` command to get further:

```
+     NATIVE_CC=$CC_FOR_BUILD \
+     NATIVE_FLAGS="-O2" \
```

**Comment by wolfv** [comment]
Going to try that. Locally I had to disable some SIMD stuff as well.

File: recipe/remove_hw_accel_osxarm64.patch [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+--- a/nss/lib/freebl/Makefile
++++ b/nss/lib/freebl/Makefile
+@@ -121,8 +121,8 @@ endif
+ endif
+ ifeq ($(CPU_ARCH),aarch64)
+     ifdef CC_IS_CLANG
+-        DEFINES += -DUSE_HW_AES -DUSE_HW_SHA1 -DUSE_HW_SHA2
+-        EXTRA_SRCS += aes-armv8.c gcm-aarch64.c sha1-armv8.c sha256-armv8.c
++        # DEFINES += -DUSE_HW_AES -DUSE_HW_SHA1 -DUSE_HW_SHA2
++        # EXTRA_SRCS += aes-armv8.c gcm-aarch64.c sha1-armv8.c sha256-armv8.c
```
Comment (isuruf) [inline recipe/remove_hw_accel_osxarm64.patch:12]: Why?

Comment (wolfv) [reply recipe/remove_hw_accel_osxarm64.patch:12]: becuase of this

```
arm64-apple-darwin20.0.0-clang -o Darwin19.6.0_arm64-apple-darwin20.0.0-clang_64_OPT.OBJ/Darwin_SINGLE_SHLIB/aes-armv8.o -c -std=c99 -O2 -fPIC  -fno-common -pipe -DDARWIN -DHAVE_STRERROR -DHAVE_BSD_FLOCK -isysroot /Applications/Xcode_12.4.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.0.sdk -Wall -Wshadow -Qunused-arguments -Wno-parentheses-equality -Wno-array-bounds -Wno-unevaluated-expression -DNSS_NO_GCC48 -DXP_UNIX -DSHLIB_SUFFIX=\"dylib\" -DSHLIB_PREFIX=\"lib\" -DSHLIB_VERSION=\"3\" -DSOFTOKEN_SHLIB_VERSION=\"3\" -DRIJNDAEL_INCLUDE_TABLES -UDEBUG -DNDEBUG -DNSS_DISABLE_AVX2 -DNSS_NO_INIT_SUPPORT -DUSE_UTIL_DIRECTLY -DNO_NSPR_10_SUPPORT -DSSL_DISABLE_DEPRECATED_CIPHER_SUITE_NAMES -DNSS_USE_64 -DUSE_HW_AES -DUSE_HW_SHA1 -DUSE_HW_SHA2 -DHAVE_INT128_SUPPORT -DMP_API_COMPATIBLE -I$PREFIX/include/nspr -I$PREFIX/include -I../../../dist/Darwin19.6.0_arm64-apple-darwin20.0.0-clang_64_OPT.OBJ/include -I../../../dist/public/nss -I../../../dist/private/nss -Impi -Iecl -Iverified -Iverified/kremlin/include -Iverified/kremlin/kremlib/dist/minimal -Ideprecated  -march=armv8-a -mfpu=crypto-neon-fp-armv8 aes-armv8.c
aes-armv8.c:14:2: error: "Compiler option is invalid"
#error "Compiler option is invalid"
 ^
aes-armv8.c:66:17: warning: implicit declaration of function 'vaeseq_u8' is invalid in C99 [-Wimplicit-function-declaration]
        state = vaeseq_u8(state, key1);
                ^
aes-armv8.c:66:15: error: assigning to 'uint8x16_t' (vector of 16 'uint8_t' values) from incompatible type 'int'
        state = vaeseq_u8(state, key1);
              ^ ~~~~~~~~~~~~~~~~~~~~~~
```

**Comment by wolfv** [comment]
thanks @xhochy for your tip! Now it fails later with a bad cpu type :/

**Comment by isuruf** [comment]
Did you check where in the Makefile it is called?

**Comment by Tobias-Fischer** [comment]
So the error is 
```
/Users/runner/miniforge3/conda-bld/nss_1618241602068/work/nss/cmd/shlibsign/./sign.sh: line 49: /Users/runner/miniforge3/conda-bld/nss_1618241602068/work/nss/cmd/shlibsign/Darwin19.6.0_arm64-apple-darwin20.0.0-clang_64_OPT.OBJ/shlibsign: Bad CPU type in executable
```

Digging in the source code of nss, sign.sh should not be called when cross-compiling:
https://github.com/nss-dev/nss/blob/master/cmd/shlibsign/Makefile#L89-L95

So we need to pass `CROSS_COMPILE=1` as argument to make, as it does not seem to be set automatically when cross-compiling on OSX (only when on Linux): https://github.com/nss-dev/nss/search?q=CROSS_COMPILE

File: recipe/build.sh [hunk 2]
```diff
@@ -5,11 +5,15 @@ cd nss
 if [[ ${HOST} =~ .*darwin.* ]]; then
     USE_GCC=0
     MACOS_SDK_DIR="${CONDA_BUILD_SYSROOT}"
+    if [[ "${target_platform}" == "osx-arm64" ]]; then
+        MACOS_CROSS_ARGS="CPU_ARCH=arm NATIVE_CC=$CC_FOR_BUILD NATIVE_FLAGS=-O2"
```
Comment (Tobias-Fischer) [inline recipe/build.sh:9]:
```suggestion
        MACOS_CROSS_ARGS="CPU_ARCH=arm NATIVE_CC=$CC_FOR_BUILD NATIVE_FLAGS=-O2 CROSS_COMPILE=1"
```

**Comment by wolfv** [comment]
Wow, thanks @Tobias-Fischer  looks like we’re almost there

**Comment by Tobias-Fischer** [comment]
I'm not sure where this is happening (`install: Darwin19.6.0_64_OPT.OBJ/lib/*.chk: No such file or directory`) - maybe in https://github.com/nss-dev/nss/blob/master/cmd/shlibsign/Makefile too?

**Comment by wolfv** [comment]
The last problem is that Make still tries to install `.chk` files, even though they are not created (since cross compile doesn't run the signtool over the shared libraries). going to try to comment that out.

**Comment by Tobias-Fischer** [comment]
Nice!

**Comment by wolfv** [comment]
@marcelotrevisani could you have a look? Seems to work fine now. I think the missing `chk` files just prevent nss to run in FIPS 140 mode which we hopefully don't need.

**Review by marcelotrevisani — APPROVED** [review]

**Pull request merged**